src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDataFileCommand.java

```diff
@@ -210,6 +213,31 @@ public FileVisitResult postVisitDirectory(final Path dir, final IOException e)
         } catch (Exception e) {
             logger.log(Level.WARNING, "Identifier deletion was not successfull:", e.getMessage());
         }
+        
+        // If there is a Map Layer associated with this file, we may need to 
+        // try and remove the layer data on the WorldMap side. 
+        if (ctxt.mapLayerMetadata().findMetadataByDatafile(doomed) != null) {
+            // (We need an AuthenticatedUser in order to produce a WorldMap token!)
+            String id = getUser().getIdentifier();
+            id = id.startsWith("@") ? id.substring(1) : id;
+            AuthenticatedUser authenticatedUser = ctxt.authentication().getAuthenticatedUser(id);
+            try {
+                ctxt.mapLayerMetadata().deleteMapLayerFromWorldMap(doomed, authenticatedUser);
+
+                // We have the dedicatd command DeleteMapLayerMetadataCommand, but 
+                // there's no need to use it explicitly, since the Dataverse-side
+                // MapLayerMetadata entity will be deleted by the database 
+                // cascade on the DataFile. -- L.A. Apr. 2020
```

Comment:
I had forgotten that there's a DeleteMapLayerMetadataCommand but this comment makes me wonder why we don't call it. I get that we don't **need** to call it thanks to the cascade, but would calling the command cause problems? Does it do more than we want? It seems to clean up thumbnails, for example. Also, there's tiny typo above: dedicatd.

Reply:
I'm still curious about these questions but there's no need to hold this up from QA so I moved it over. This pull request adds a lot of value.

Reply:
Sorry, missed the original question 4 hours ago. I guess the answer is, the delete layer command is a more surgical tool; it's for a case where we just need to delete the layer, without touching the file. 
Calling it in the context of trying to wipe everything, including the datafile and all its associated physical files, would not hurt, strictly speaking - but I couldn't think of why we would want to call it either; seeing how the cascade is moire reliable? It does remove the map thumbnails - but then we run a full physical file cleanup after DeleteDataFileCommand, that will delete all aux files associated with the datafile, thumbnails and otherwise. 
And then again there is the fact the command doesn't do everything we need, either. It does not attempt to go out and remove whatever possible data may be sitting on the WorldMap side. I could only *assume* that it was by design. 
My attitude with this PR was to check in something ASAP that works and doesn't break things. 
I generally sympathize with the goals of making things cleaner; calling commands instead of database queries and such. But in the case of WorldMap, the whole is a big hack - isn't it? Our attempt to "think about it some more" to see if we were doing everything right resulted in a 2.5 year delay, and a whole bunch of these un-destructible datasets still sitting in production...

Reply:
Yes, agreed. I don't think this will break anything and it definitely adds value. Thanks for the explanation.